lib/rules/boolean-prop-naming.js

```diff
@@ -240,21 +240,25 @@ module.exports = {
       }
 
       if (
-        component.node.parent
-        && component.node.parent.type === 'VariableDeclarator'
-        && component.node.parent.id
-        && component.node.parent.id.type === 'Identifier'
-        && component.node.parent.id.typeAnnotation
-        && component.node.parent.id.typeAnnotation.typeAnnotation
-        && component.node.parent.id.typeAnnotation.typeAnnotation.typeParameters
-        && (
-          component.node.parent.id.typeAnnotation.typeAnnotation.typeParameters.type === 'TSTypeParameterInstantiation'
-          || component.node.parent.id.typeAnnotation.typeAnnotation.typeParameters.type === 'TypeParameterInstantiation'
-        )
+        !component.node.parent
+        || component.node.parent.type !== 'VariableDeclarator'
```

Comment:
must we invert the condition? that makes for a noisier diff.

Reply:
I want to exit early here. I could invert the expression with `!(...statement...)` to have a smaller diff, but that feels dirty. Thoughts?

Reply:
it'd probably be easier to read the diff with a single commit that inverts the condition, but changes nothing else; and then the rest of this PR on top of it?

Reply:
Ah I understand, this commit is very close to doing that; https://github.com/jsx-eslint/eslint-plugin-react/pull/3629/commits/0f45d976eb6f97b2cad0e1f72f2a802ea3aab657

Reply:
If it's helpful, I can open smaller prs for each of the files touched?

Reply:
sure, that works too - or maybe one PR containing solely refactors (that don't change any behavior). then i can land that, and this can be rebased on top of it.